### DISCUSSION TITLE
Run Hours and Fuel Consumption not updating on Maintenance page

### DISCUSSION THREAD

**Discussion by SRFurst** [jgyates/genmon#1114] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

1.19.02

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

Fuel Consumption and Run Hours do not seem to be updating for me on the Maintenance page. The ambient temperature IS updating. I ran the generator for four hours (manually with no load) and there was no change in run hours. Am I misunderstanding how Run Hours works? Does it only record run time under load (utility power failure?).

Controller Detected : Evolution 2.0, Air Cooled
    Nominal RPM : 3600
    Rated kW : 22 kW
    Nominal Frequency : 60 Hz
    Fuel Type : Natural Gas
    Generator Phase : 1
    Engine Displacement : 999 cc
    kW Hours in last 30 days : 0.00 kWh
    Fuel Consumption in last 30 days : 12.92 cubic feet
    Total Power Log Fuel Consumption : 19.73 cubic feet
    Run Hours in last 30 days : 0.10 h
    Run Hours in the last year : 0.15 h
    Ambient Temperature Sensor : 57 F

**Comment by jgyates** [comment, marked as answer]
The Run hours over a period of time and fuel consumption calculations are all based on the power log that genmon keeps. The power log will log the power used by the generator. For Evolution liquid cooled models, power is calculated from the leg one and leg two current (which are read over modbus from the built in current transformers). For other models the power is derived from other sources (direct modbus read, Hall Effect Sensor, etc) or it is not supported.

The power log is not updated if the controller is in the off or stopped state. The power log is not updated if the power value has not changed since the last reading. 

You should be able to look at the status page and see the L1 and L2 current. If these values are updating you should see entries in your power log (/etc/genmon/kwlog.txt) and this should result in readings updated on the Maintenance page.

I suspect that your current transformers are reading zero or close to zero as you are not running on load. The current transformers will measure zero current if the load is zero. You should be able to see the output of the current transformers on the Status page to validate this. If this is true the calculation of power for zero current would be zero. A zero entry in the power log denotes no run hours.

**Reply by SRFurst** [reply]
Thank you for your very detailed explanation. I guess my concern is that, from a maintenance standpoint, I want to know how many hours my generator has run (exercise+utility outage+other). I understand that NG consumption will depend on load and that calculating this in a no-load situation would be difficult. But I still want to know how many hours my gennie has run, preferably without adding up times from the Log, so I can schedule oil changes. (Yes, I could actually walk over to the generator and look at the panel but what fun is that?)

**Reply by jgyates** [reply]
The total hours your generator engine has run is maintained by the controller and displayed by genmon. The total engine run hours are maintained by the controller and available lower down on the maintenance page (under the Maintenance -> Service -> Total Run Hours heading). This value is read directly from the controller. Also, the maintenance timers for service intervals are also displayed lower down on the maintenance page (hours remaining until servce and date service is due, which ever comes first). These values are also maintained by the controller. Once the service timer expires the controller will generate a Service Due alarm, which genmon will detect and relay via your preferred notification method.

The power log is not designed to provide information on maintenance intervals as the controller already provides this functionality. It's original intent is to provide people with tanks (propane and diesel) an estimate of the fuel used and the amount of fuel remaining in the tank. when they fill the tank they have to reset the log to get an accurate estimate of fuel remaining in the tank. Once this functionality was added it was expanded to support natural gas (e.g. not tank) configurations.

Let me know if you have any other questions.

**Reply by SRFurst** [reply]
Thank you so much!
